cli-tests/src/upload.rs

```diff
@@ -223,14 +225,24 @@ async fn upload_bundle_using_bep() {
 
     let tar_extract_directory = assert_matches!(&requests[3], RequestPayload::S3Upload(d) => d);
 
-    let file = fs::File::open(tar_extract_directory.join("junit/0")).unwrap();
-    let reader = BufReader::new(file);
+    let junit_file = fs::File::open(tar_extract_directory.join("junit/0")).unwrap();
+    let junit_reader = BufReader::new(junit_file);
 
     // Uploaded file is a junit, even when using BEP
     let mut junit_parser = JunitParser::new();
-    assert!(junit_parser.parse(reader).is_ok());
+    assert!(junit_parser.parse(junit_reader).is_ok());
     assert!(junit_parser.errors().is_empty());
 
+    let mut bazel_bep_parser = BazelBepParser::new(
+        tar_extract_directory
+            .join("bazel_bep.json")
+            .to_string_lossy()
+            .to_string(),
+    );
```

Comment:
If we update `BazelBepParser` you won't have to do all of this `Path` <-> `String` conversion:

```rust
#[derive(Debug, Clone, Default)]
pub struct BazelBepParser {
    bazel_bep_path: PathBuf,
    test_results: Vec<TestResult>,
    bep_test_events: Vec<BuildEvent>,
    errors: Vec<String>,
}

impl BazelBepParser {
    pub fn new<T: Into<PathBuf>>(bazel_bep_path: T) -> Self {
        Self {
            bazel_bep_path: bazel_bep_path.into(),
            ..Default::default()
        }
    }
    // ... other code ...
}
```

Reply:
Everytime I'm sitting here thinking "there has to be a better way," Dylan's got me